packages/pkl.lua/lua.pkl

```diff
@@ -0,0 +1,283 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// A renderer for [Lua](https://www.lua.org) configuration files.
+@ModuleInfo { minPklVersion = "0.25.0" }
+module pkl.lua.lua
+
+import "pkl:reflect"
+
+local const pathSpecRegex: Regex = let (prop = #"(?:[^\[\]^*.]+)"#) Regex(#"""
+  (?x)
+  \^?
+  (?:
+    (?:
+      \#(prop)
+    | \*
+    | \[(?:\#(prop)|\*)\]
+    )
+    (?:
+      \.\#(prop)
+    | \.\*
+    | \[(?:\#(prop)|\*)\]
+    )*
+  )?
+  """#)
+local const pathSpecSplitRegex: Regex = Regex(#"\.|(?=\[)|(?<=\^|\])"#)
+
+local class Prop {
+  name: String
+}
+local class Key {
+  key: String
+}
+
+local typealias PathEntry = Prop|Key|"^"
+
+local const splatKey: Key = new Key { key = "*" }
+
+/// A string that is a Lua reserved keyword.
+///
+/// These strings are not allowed to be used as identifiers.
+typealias LuaKeyword = "and"|"break"|"do"|"else"|"elseif"|"end"|"false"|"for"|"function"|"goto"|"if"|"in"|"local"|"nil"|"not"|"or"|"repeat"|"return"|"then"|"true"|"until"|"while"
+
+/// A string that is a valid Lua identifier.
+typealias LuaIdentifier = String(matches(Regex("[a-zA-Z_][a-zA-Z0-9_]*")) && !(this is LuaKeyword))
+
+/// Directs [Renderer] to output additional text [before] and/or [after] rendering a [value].
+typealias LuaRenderDirective = PcfRenderDirective
```

Comment:
I ended up replacing it with its own class already. And the intended use-case here is to be able to emit code that operates on values instead of just emitting values. For example, you could have a converter like `[MyClass] = (it) -> new LuaRenderDirective { before = "setmetatable(", value = it, after = ", classMetatable)" }`. Such a converter would cause `MyClass` to be emitted as a table with a particular metatable attached.